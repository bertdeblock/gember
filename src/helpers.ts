import type { EmberPackageJson } from "./types.js";

export function isV1Addon(packageJson: EmberPackageJson): boolean {
  if (isAddon(packageJson)) {
    const { version } = packageJson["ember-addon"] ?? {};

    return version === 1 || version === undefined;
  }

  return false;
}

export function isV2Addon(packageJson: EmberPackageJson): boolean {
  if (isAddon(packageJson)) {
    const { version } = packageJson["ember-addon"] ?? {};

    return version === 2;
  }

  return false;
}

function isAddon(packageJson: EmberPackageJson): boolean {
  if (Array.isArray(packageJson.keywords)) {
    return packageJson.keywords.includes("ember-addon");
  }

  return false;
}
