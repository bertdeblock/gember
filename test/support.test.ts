import { afterEach, it } from "vitest";
import { Package } from "./helpers.ts";

let pkg: Package;

afterEach(() => pkg.cleanUp());

it("supports v1 apps", async (ctx) => {
  pkg = await Package.create("v1-app");

  await pkg.gember("component", "foo");

  const content = await pkg.readFile("app/components/foo.gjs");

  ctx.expect(content).toMatchSnapshot();
});

it("supports v2 apps", async (ctx) => {
  pkg = await Package.create("v2-app");

  await pkg.gember("component", "foo");

  const content = await pkg.readFile("app/components/foo.gjs");

  ctx.expect(content).toMatchSnapshot();
});

it("supports v1 addons", async (ctx) => {
  pkg = await Package.create("v1-addon");

  await pkg.gember("component", "foo");

  const content = await pkg.readFile("addon/components/foo.gjs");

  ctx.expect(content).toMatchSnapshot();
});

it("supports v2 addons", async (ctx) => {
  pkg = await Package.create("v2-addon");

  await pkg.gember("component", "foo");

  const content = await pkg.readFile("src/components/foo.gjs");

  ctx.expect(content).toMatchSnapshot();
});
