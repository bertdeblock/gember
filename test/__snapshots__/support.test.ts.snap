// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`supports v1 addons 1`] = `
"<template>{{yield}}</template>
"
`;

exports[`supports v1 apps 1`] = `
"<template>{{yield}}</template>
"
`;

exports[`supports v2 addons 1`] = `
"<template>{{yield}}</template>
"
`;

exports[`supports v2 addons 2`] = `
"import { render } from '@ember/test-helpers';
import { module, test } from 'qunit';
import Foo from 'v2-addon/components/foo';
import { setupRenderingTest } from 'ember-qunit';

module('Integration | Component | Foo', function (hooks) {
  setupRenderingTest(hooks);

  test('it renders', async function (assert) {
    await render(<template><Foo /></template>);

    assert.dom().hasText('');

    await render(
      <template>
        <Foo>
          template block text
        </Foo>
      </template>
    );

    assert.dom().hasText('template block text');
  });
});
"
`;

exports[`supports v2 apps 1`] = `
"<template>{{yield}}</template>
"
`;
